server/channels/app/plugin_requests.go

```diff
@@ -158,96 +188,73 @@ func (ch *Channels) servePluginRequest(w http.ResponseWriter, r *http.Request, h
 	}
 
-	// Mattermost-Plugin-ID can only be set by inter-plugin requests
-	r.Header.Del("Mattermost-Plugin-ID")
+	csrfCheckPassed := false
+	if (session != nil && session.Id != "") && appErr == nil && cookieAuth && r.Method != "GET" {
+		sentToken := ""
 
-	r.Header.Del("Mattermost-User-Id")
-	if token != "" {
-		csrfCheckPassed := false
-		if (session != nil && session.Id != "") && appErr == nil && cookieAuth && r.Method != "GET" {
-			sentToken := ""
-
-			if r.Header.Get(model.HeaderCsrfToken) == "" {
-				bodyBytes, _ := io.ReadAll(r.Body)
-				r.Body = io.NopCloser(bytes.NewBuffer(bodyBytes))
-				if err := r.ParseForm(); err != nil {
-					mlog.Warn("Failed to parse form data for plugin request", mlog.Err(err))
-				}
-				sentToken = r.FormValue("csrf")
-				r.Body = io.NopCloser(bytes.NewBuffer(bodyBytes))
-			} else {
-				sentToken = r.Header.Get(model.HeaderCsrfToken)
+		if r.Header.Get(model.HeaderCsrfToken) == "" {
+			bodyBytes, _ := io.ReadAll(r.Body)
+			r.Body = io.NopCloser(bytes.NewBuffer(bodyBytes))
+			if err := r.ParseForm(); err != nil {
+				mlog.Warn("Failed to parse form data for plugin request", mlog.Err(err))
 			}
+			sentToken = r.FormValue("csrf")
+			r.Body = io.NopCloser(bytes.NewBuffer(bodyBytes))
+		} else {
+			sentToken = r.Header.Get(model.HeaderCsrfToken)
+		}
 
-			expectedToken := session.GetCSRF()
-
-			if sentToken == expectedToken {
-				csrfCheckPassed = true
-			}
+		expectedToken := session.GetCSRF()
 
-			// ToDo(DSchalla) 2019/01/04: Remove after deprecation period and only allow CSRF Header (MM-13657)
-			if r.Header.Get(model.HeaderRequestedWith) == model.HeaderRequestedWithXML && !csrfCheckPassed {
-				csrfErrorMessage := "CSRF Check failed for request - Please migrate your plugin to either send a CSRF Header or Form Field, XMLHttpRequest is deprecated"
-				sid := ""
-				userID := ""
-
-				if session.Id != "" {
-					sid = session.Id
-					userID = session.UserId
-				}
-
-				fields := []mlog.Field{
-					mlog.String("path", r.URL.Path),
-					mlog.String("ip", r.RemoteAddr),
-					mlog.String("session_id", sid),
-					mlog.String("user_id", userID),
-				}
-
-				if *ch.cfgSvc.Config().ServiceSettings.ExperimentalStrictCSRFEnforcement {
-					mlog.Warn(csrfErrorMessage, fields...)
-				} else {
-					mlog.Debug(csrfErrorMessage, fields...)
-					csrfCheckPassed = true
-				}
-			}
-		} else {
+		if sentToken == expectedToken {
 			csrfCheckPassed = true
 		}
 
-		if (session != nil && session.Id != "") && appErr == nil && csrfCheckPassed {
-			r.Header.Set("Mattermost-User-Id", session.UserId)
-			context.SessionId = session.Id
+		// ToDo(DSchalla) 2019/01/04: Remove after deprecation period and only allow CSRF Header (MM-13657)
+		if r.Header.Get(model.HeaderRequestedWith) == model.HeaderRequestedWithXML && !csrfCheckPassed {
+			csrfErrorMessage := "CSRF Check failed for request - Please migrate your plugin to either send a CSRF Header or Form Field, XMLHttpRequest is deprecated"
```

Comment:
I guess we do all the work above to log this today all the time, even though it's almost never enforced. Given this method of proactive logging hasn't proven effective (I just ignore the logs myself), would we consider refactoring to check `ServiceSettings.ExperimentalStrictCSRFEnforcement` upfront and only then do the work to figure it all out?

Reply:
(I also know we're looking to enable this by default for new customers in v11, cc @enzowritescode.)